src/core/public/application/application_service.tsx

```diff
@@ -243,6 +243,9 @@ export class ApplicationService {
       throw new Error('ApplicationService#setup() must be invoked before start.');
     }
 
+    const httpLoadingCount$ = new BehaviorSubject(0);
+    http.addLoadingCountSource(httpLoadingCount$);
```

Comment:
It's better than what we had before, which was nothing, but is this really sufficient? Don't we want a more visible, 'full page' loader (such as the initial loader, or a full page EuiLoadingSpinner) while awaiting the app to load?

`http.loadingCount` was meant to have a visual notification of 'background' requests. I think awaiting for the app to load could benefit from a more visible indicator. What I had in mind was more adapting `app_container.tsx` to show an in-page loader while awaiting for the mount function.

But https://github.com/elastic/kibana/issues/50938 do state
> We can probably reuse the existing loading bar indicator that shows at the top of the page.

So I guess this is fine.